frictionless/errors/row.py

```diff
@@ -66,11 +66,72 @@ class PrimaryKeyError(RowError):
     template = 'Row at position "{rowNumber}" violates the primary key: {note}'
 
 
-class ForeignKeyError(RowError):
+@attrs.define(kw_only=True)
+class ForeignKeyError(TableError):
     type = "foreign-key"
     title = "ForeignKey Error"
     description = "Values in the foreign key fields should exist in the reference table"
     template = 'Row at position "{rowNumber}" violates the foreign key: {note}'
+    tags = ["#table", "#row"]
+
+    cells: List[str]
+    """NOTE: add docs"""
+
+    row_number: int
+    """NOTE: add docs"""
+
+    target_keys: List[str]
+    """NOTE: add docs"""
+
+    source_keys: List[int]
+    """NOTE: add docs"""
+
+    source_name: List[str]
+    """NOTE: add docs"""
+
+    missing_values: List[int]
+    """NOTE: add docs"""
+
+    @classmethod
+    def from_row(
+        cls, row, *, target_keys, source_keys, source_name, missing_values, note
+    ):
+        """Create an foreign-key-error from a row
+
+        Parameters:
+            row (Row): row
+            target_keys (tuple): target keys
+            source_keys (tuple): source keys
+            source_name: (str): source name
+            missing_values (tuple): missing values
+            note (str): note
+
+        Returns:
+            ForeignKeyError: error
+        """
+        to_str = lambda v: str(v) if v is not None else ""
+        return cls(
+            note=note,
+            cells=list(map(to_str, row.cells)),
+            row_number=row.row_number,
+            target_keys=target_keys,
+            source_keys=source_keys,
+            source_name=source_name,
+            missing_values=missing_values,
+        )
+
+    # Metadata
+
+    metadata_profile_patch = {
+        "properties": {
+            "cells": {"type": "array", "items": {"type": "string"}},
```

Comment:
I think `cells` and `rowNumber` are also inherited from the RowError class (to be checked)